File: javascript/two pointers/0011. Container With Most Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let res=0,curr=0
    let i=0
    let j=height.length-1
    while(i<j){
        curr=(j-i)*Math.min(height[i],height[j])
        res=Math.max(res,curr)
        if(height[i]<=height[j])i++;
        else j--;
    }
    return res;
};

File: NeetCode/C++/two pointers/0011. Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int res,curr=0;
        int i=0;
        int j=height.size()-1;
        while(i<j){
            curr=(j-i)*min(height[i],height[j]);
            res=max(curr,res);
            if(height[j]>=height[i])i++;
            else j--;
        }
        return res;
    }
};
//time complexity : O(N)